WordPress/Classes/Models/AbstractPost+Searchable.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+extension AbstractPost: SearchableItemConvertable {
+    var searchItemType: SearchItemType {
+        return .abstractPost
+    }
+
+    var isSearchable: Bool {
+        guard status != .trash else {
+            // Don't index trashed posts
+            return false
+        }
+        guard blog.visible else {
+            // Don't index posts for non-visible sites (not visible in the My Sites list)
+            return false
+        }
+        return true
+    }
+
+    var searchIdentifier: String? {
+        guard let postID = postID, postID.intValue > 0 else {
+            return nil
+        }
+        return postID.stringValue
+    }
+
+    var searchDomain: String? {
+        if let dotComID = blog.dotComID, dotComID.intValue > 0 {
+            return dotComID.stringValue
+        } else {
+            // This is a self-hosted site, set domain to the xmlrpc string
+            return blog.xmlrpc
+        }
+    }
+
+    var searchTitle: String? {
+        return generateTitle(from: postTitle)
+    }
+
+    var searchDescription: String? {
+        guard let postPreview = contentPreviewForDisplay(), !postPreview.isEmpty else {
+            return blog.displayURL as String? ?? contentForDisplay()
+        }
+        return postPreview
+    }
+
+    var searchKeywords: [String]? {
+        return generateKeywordsFromContent()
+    }
+}
+
+// MARK: - Private Helper Functions
+
+fileprivate extension AbstractPost {
+    func generateKeywordsFromContent() -> [String]? {
+        var keywords: [String]? = nil
+        if let postTitle = postTitle {
+            // Try to generate some keywords from the title...
+            keywords = postTitle.components(separatedBy: " ").map({ $0.trimmingCharacters(in: .whitespacesAndNewlines) })
```

Comment:
Possible future optimisation: I wonder if we should use `NSLinguisticTagger` here instead of splitting on spaces – Apple's own documentation states:

> Using NSLinguisticTagger to enumerate words, rather than simply splitting components by whitespace, ensures correct behavior in multiple scripts and languages. For example, neither Chinese nor Japanese uses spaces to delimit words.

Reply:
Heh. @frosty great minds... 😄  I actually have a version of that working, but `NSLinguisticTagger` is very slow. I'm thinking of a future PR doing the heavy lifting on a background thread perhaps.... 🤔